### ISSUE TITLE
Example of Mincut Clustering

### ISSUE THREAD

**Issue by jlevy44** [rusty1s/pytorch_geometric#967]
Hey! Can you add an example of clustering with mincut? Can't seem to get my code to work here. It has only been predicting one label for some reason.

**Comment by rusty1s** [comment]
There is an example called proteins_mincut.

**Comment by jlevy44** [comment]
Yeah I’ve seen that example, but it is somewhat supervised, yes? I’m just wondering how it has fared running only the unsupervised loss functions. I’m trying to execute this on one graph as a use case

**Comment by jlevy44** [comment]
It was a very helpful example, I’m just trying to understand why I only get one cluster on my datasets rather than multiple when I run an argmax after training

**Comment by jlevy44** [comment]
okay I was able to get it working, just your traditional convergence issues.

If you're node features are missing, is it possible to initialize the node features with laplacian eigenmaps, or is that "cheating"? I was able to get the algo to converge with a spectral embedding initialization when node attributes were missing versus inputting a "one" matrix/vector. Thanks!

**Comment by rusty1s** [comment]
I wouldn't call that cheating. Seems perfectly reasonable to me :)

**Comment by jlevy44** [comment]
My one concern then is how much of the learning is GCN and how much is the spectral embedding. Else the method could be entirely dependent on what spectral embedding does, which could be disadvantageous if confronted with a graph of higher order relationships

**Comment by rusty1s** [comment]
You should evaluate if the GCN brings any gain, *e.g.*, by using an MLP on your spectral embedding as a baseline.

**Comment by jlevy44** [comment]
yeah, comparisons to Spectral Clustering will be performed, but I think more importantly I'm trying to compare to other CD algorithms... Fuzzy clustering  is definitely a plus though

**Comment by jlevy44** [comment]
I wonder if SC and minut could be set up in a transfer learning / teacher student sense though. Learn SC using the GCN and then fine tune, would be interesting. What I’m proposing is almost equivalent to that, but as if I froze layers before computation of the SC.
